linked_list/implementation.py

```diff
@@ -1,40 +1,127 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        if elements:
+            for e in elements:
+                node = Node(e)
+                if not self.start:
+                    self.start = node
+                if self.end:
+                    self.end.next = node
+                self.end = node
 
     def __str__(self):
-        pass
+        str_list = []
+        for i in iter(self):
+            str_list.append(i.elem)
+        return str(str_list)
 
     def __len__(self):
-        pass
+        total = 0
+        for item in iter(self):
+            total += 1
+        return total
 
     def __iter__(self):
-        pass
+        if self.start:
+            element = self.start
+            while element.next:
+                yield element
+                element = element.next
+            # yield end element
+            yield element
```

Comment:
I like your code. I'm just wondering something (and please correct me if I'm wrong). Instead of jumping over the `while` loop to return the last `element`, Wouldn't it be simpler/clearer to do:
```python
while element:
    yield element
```
And maybe raise a `StopIteration` or something?